nixos/modules/services/games/xonotic.nix

```diff
@@ -0,0 +1,130 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) mkEnableOption mkIf mkOption types;
+
+  cfg = config.services.xonotic;
+
+  toCfg = lib.generators.toKeyValue {
+    mkKeyValue = lib.generators.mkKeyValueDefault {
+      mkValueString = v:
+        if v == true then
+          "1"
+        else if v == false then
+          "0"
+        else if builtins.isString v then
+          ''"${v}"''
+        else
+          lib.generators.mkValueStringDefault { } v;
+    } " ";
+  };
+
+  toConfig = config: toCfg config;
+
+  writeConfig = { config, extraConfig }:
+    pkgs.writeText "server.cfg" ''
+      ${toConfig config}
+      ${extraConfig}
+    '';
+
+  serverModule = types.submodule ({ name, ... }: {
+    options = {
+      config = mkOption {
+        type = types.attrsOf (types.oneOf [ types.str types.int types.bool ]);
+        description = ''
+          server config, see
+          <link xlink:href="https://github.com/xonotic/xonotic/wiki/basic-server-configuration">https://github.com/xonotic/xonotic/wiki/basic-server-configuration</link>
```

Comment:
I'm not very good with our documentation format but I *thought* you could avoid duplicating the link and it will show up the same way.